esp_delta_ota/examples/https_delta_ota/tools/esp_delta_ota_patch_gen.py

```diff
@@ -19,40 +23,120 @@
+        patch_file.write(crc32.to_bytes(CRC_SIZE, 'little'))
         patch_file.write(bytearray(RESERVED_HEADER))
-        with open(patch_file_without_header, "rb") as temp_patch:
-            patch_file.write(temp_patch.read())
+        patch_file.write(patch_content)
 
     os.remove(patch_file_without_header)
 
-def main() -> None:
-    parser = argparse.ArgumentParser('Delta OTA Patch Generator Tool')
-    parser.add_argument('--chip', help="Target", default="esp32")
-    parser.add_argument('--base_binary', help="Path of Base Binary for creating the patch")
-    parser.add_argument('--new_binary', help="Path of New Binary for which patch has to be created")
-    parser.add_argument('--patch_file_name', help="Patch file path", default="patch.bin")
+# This api applies the patch file over base_binary file and generates the binary.new file. Then it compares 
+# the hash of new_binary and binary.new, if they are same then verfications get successfull otherwise fails.
+def verify_patch(base_binary: str, patch_to_verify: str, new_binary: str) -> None:
+
+    with open(patch_to_verify, "rb") as original_file:
+        # Read CRC32 from header
+        original_file.seek(MAGIC_SIZE + DIGEST_SIZE + HEADER_VERSION_SIZE)
+        header_crc32 = int.from_bytes(original_file.read(CRC_SIZE), 'little')
+        
+        # Calculate CRC32 of patch content
+        original_file.seek(HEADER_SIZE)  # Move the file pointer 64 bytes ahead
+        patch_content = original_file.read()  # Read the rest of the file
+        calculated_crc32 = zlib.crc32(patch_content)
+        
+        # Check if CRC32 matches
+        if header_crc32 != calculated_crc32:
+            print("CRC32 mismatch. Patch file may be corrupted.")
+            return
 
-    args = parser.parse_args()
+    with tempfile.NamedTemporaryFile() as temp_file:
+        temp_file.write(patch_content)
+        temp_file.flush()
 
-    create_patch(args.chip, args.base_binary, args.new_binary, args.patch_file_name)
+        cmd = "detools apply_patch " + base_binary + " " + temp_file.name + " binary.new"
+
+        try:
+            proc = subprocess.Popen([cmd], stdout=subprocess.PIPE, shell = True)
+        except:
+            print("Failed to execute the command `detools apply_patch`")
+            return
+            
+        (out, err) = proc.communicate()
+        x = re.search(b"Successfully created" ,out) 
+        if x :
+            sha_of_new_created_binary = calculate_sha256("binary.new")
+            sha_of_new_binary = calculate_sha256(new_binary)
+            
+            if(sha_of_new_created_binary == sha_of_new_binary):
+                print("Patch file verified successfully")
+            else:
+                print("Failed to verify the patch")
+        else:
+            print("Failed to verify the patch")
+        os.remove("binary.new")
+
+def main() -> None:
+    if len(sys.argv) < 2:
+        print("Usage: python file_name.py create_patch/verify_patch [arguments]")
```

Comment:
Please put a name of the utility script itself here `esp_delta_ota_patch_gen.py`